src/compressor/zlib/ZlibCompressor.cc

```diff
@@ -93,20 +99,88 @@ int ZlibCompressor::compress(const bufferlist &in, bufferlist &out)
   return 0;
 }
 
+int ZlibCompressor::isal_compress(const bufferlist &in, bufferlist &out)
+{
+#if (__arm__)
+  derr << "Unsupported architecture" << dendl;
+  return -1;
+#else
+  int ret;
+  unsigned have;
+  isal_zstream strm;
+  unsigned char* c_in;
+  int begin = 1;
+
+  /* allocate deflate state */
+  isal_deflate_init(&strm);
+  strm.end_of_stream = 0;
+
+  for (std::list<buffer::ptr>::const_iterator i = in.buffers().begin();
+      i != in.buffers().end();) {
+
+    c_in = (unsigned char*) (*i).c_str();
+    long unsigned int len = (*i).length();
+    ++i;
+
+    strm.avail_in = len;
+    strm.end_of_stream = (i == in.buffers().end());
+    strm.flush = FINISH_FLUSH;
+
+    strm.next_in = c_in;
+
+    do {
+      strm.avail_out = max_len;
+      bufferptr ptr = buffer::create_page_aligned(max_len);
+      strm.next_out = (unsigned char*)ptr.c_str() + begin;
+      if (begin) {
+        ptr.c_str()[0] = 1;
+        begin = 0;
+      }
+      ret = isal_deflate(&strm);
+      if (ret != COMP_OK) {
+         dout(1) << "Compression error: isal_deflate return error ("
+              << ret << ")" << dendl;
+         return -1;
+      }
+      have = max_len - strm.avail_out;
+      out.append(ptr, 0, have);
+    } while (strm.avail_out == 0);
+    if (strm.avail_in != 0) {
+      dout(10) << "Compression error: unused input" << dendl;
+      return -1;
+    }
+  }
+
+  return 0;  
+#endif
+}
+
+int ZlibCompressor::compress(const bufferlist &in, bufferlist &out)
+{
+  if (isal_enabled)
+    return isal_compress(in, out);
```

Comment:
I forget the exact reason for failure; maybe it was that isal_deflate was still called but not defined?  in any event, my branch has a different approach that works